libmamba/src/specs/regex_spec.cpp

```diff
@@ -39,25 +38,48 @@ namespace mamba::specs
     }
 
     RegexSpec::RegexSpec()
-        : RegexSpec(std::regex(free_pattern.data(), free_pattern.size()), std::string(free_pattern))
+        : RegexSpec(std::string(free_pattern))
     {
     }
 
-    RegexSpec::RegexSpec(std::regex pattern, std::string raw_pattern)
-        : m_pattern(std::move(pattern))
-        , m_raw_pattern(std::move(raw_pattern))
+    RegexSpec::RegexSpec(std::string raw_pattern)
     {
+        // Construct ss from raw_pattern, in particular make sure to replace all `*` by `.*`
+        // in the pattern if they are not preceded by a `.`.
         // We force regex to start with `^` and end with `$` to simplify the multiple
         // possible representations, and because this is the safest way we can make sure it is
         // not a glob when serializing it.
-        if (!util::starts_with(m_raw_pattern, pattern_start))
-        {
-            m_raw_pattern.insert(m_raw_pattern.begin(), pattern_start);
-        }
-        if (!util::ends_with(m_raw_pattern, pattern_end))
+        std::ostringstream ss;
+        ss << pattern_start;
+
+        auto first_character_it = raw_pattern.cbegin();
+        auto last_character_it = raw_pattern.cend() - 1;
+
+        for (auto it = first_character_it; it != raw_pattern.cend(); ++it)
         {
-            m_raw_pattern.push_back(pattern_end);
+            if (it == first_character_it && *it == pattern_start)
```

Comment:
Looks like this `if` and the following have the same first conditoin, so it's more like

```c++
if (it == first_character_it && (*it == pattern_start || *it == pattern_end))
{
    continue;
}
```

Reply:
The statement look similar, but the second one is using another condition (i.e. `it == last_character_it`).